nilearn/connectivity/embedding.py

```diff
@@ -0,0 +1,394 @@
+import warnings
+from math import floor, sqrt
+
+import numpy as np
+from scipy import linalg
+
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.covariance import EmpiricalCovariance
+from .._utils.extmath import is_spd
+
+
+def _check_mat(mat, prop):
+    """Raise a ValueError if the input matrix does not satisfy the property.
+
+    Parameters
+    ----------
+    mat : numpy.ndarray
+        Input array.
+
+    prop : {'square', 'symmetric', 'spd'}
+        Property to check.
+    """
+    if prop == 'square':
+        if mat.ndim != 2 or (mat.shape[0] != mat.shape[-1]):
+            raise ValueError('Expected a square matrix, got array of shape' +
+                             ' {0}.'.format(mat.shape))
+    if prop == 'symmetric':
+        if not np.allclose(mat, mat.T):
+            raise ValueError('Expected a symmetric matrix.')
+
+    if prop == 'spd':
+        if not is_spd(mat, decimal=7):
+            raise ValueError('Expected a symmetric positive definite matrix.')
+
+
+def _map_eig(function, vals, vecs):
+    """Return the symmetric matrix with eigenvectors vecs and eigenvalues
+    obtained by applying the function to vals.
+
+    Parameters
+    ----------
+    function : function
+        The function to apply.
```

Comment:
Please be more precise: this is a scalar function that takes non-negative inputs...